src/group_theory/submonoid/membership.lean

```diff
@@ -233,12 +233,19 @@ lemma pow_right_injective_iff_pow_injective {n : M} :
   function.injective (λ m : ℕ, n ^ m) ↔ function.injective (pow n) :=
 subtype.coe_injective.of_comp_iff (pow n)
 
-theorem log_pow_eq_self [decidable_eq M] {n : M} (h : function.injective (λ m : ℕ, n ^ m)) (m : ℕ) :
-  log (pow n m) = m :=
-pow_right_injective_iff_pow_injective.mp h $ pow_log_eq_self _
+/-- The exponentiation map is an isomorphism from the additive monoid on natural numbers to powers
+when it is injective. The inverse is given by the logarithms. -/
+@[simps]
+def pow_log_equiv [decidable_eq M] {n : M} (h : function.injective (λ m : ℕ, n ^ m)):
+  multiplicative ℕ ≃* powers n :=
+{ to_fun := pow n,
```

Comment:
This one is also misleading, and should be
```suggestion
{ to_fun := λ m, pow n m.to_add,
```

Reply:
Getting these right is important, otherwise you end up with nonsense goals like `1*1=2` where the `*` is really `+`!